File: blog-web-app/csharp/tests/BlogWebApp.Tests/FixedClock.cs
using BlogWebApp;

namespace BlogWebApp.Tests;

public class FixedClock : IClock
{
    private DateTime _current;

    public FixedClock(DateTime current)
    {
        _current = current;
    }

    public DateTime Now() => _current;

    public void AdvanceTo(DateTime dateTime) => _current = dateTime;

    public void AdvanceByMinutes(int minutes) => _current = _current.AddMinutes(minutes);
}

File: parking-lot/typescript/tests/FixedClock.ts
import type { Clock } from '../src/Clock.js';

export class FixedClock implements Clock {
  private current: Date;

  constructor(start: Date) {
    this.current = start;
  }

  now(): Date {
    return this.current;
  }

  advanceTo(when: Date): void {
    this.current = when;
  }

  advanceMs(ms: number): void {
    this.current = new Date(this.current.getTime() + ms);
  }

  advanceHours(hours: number): void {
    this.advanceMs(hours * 3_600_000);
  }

  advanceMinutes(minutes: number): void {
    this.advanceMs(minutes * 60_000);
  }
}
